File: crm-frontend/src/services/projectService.js
import api from './api'

const unwrap = (response) => response?.data ?? response

export const projectService = {
  // Get all projects for a workspace
  getAll: async (workspaceId) => {
    const response = await api.get(`/projects/workspace/${workspaceId}`)
    return unwrap(response)
  },

  // Get project by ID
  getById: async (id) => {
    const response = await api.get(`/projects/${id}`)
    return unwrap(response)
  },

  // Create new project — POST /api/projects with workspaceId in body
  create: async (workspaceId, projectData) => {
    const response = await api.post('/projects', { ...projectData, workspaceId: Number(workspaceId) })
    return unwrap(response)
  },

  // Update project
  update: async (id, projectData) => {
    const response = await api.put(`/projects/${id}`, projectData)
    return unwrap(response)
  },

  // Delete project
  delete: async (id) => {
    const response = await api.delete(`/projects/${id}`)
    return unwrap(response)
  },

  // Get project members
  getMembers: async (id) => {
    const response = await api.get(`/projects/${id}/members`)
    return unwrap(response)
  },

  // Add member to project
  addMember: async (id, memberData) => {
    const response = await api.post(`/projects/${id}/members`, memberData)
    return unwrap(response)
  },

  // Remove member from project
  removeMember: async (projectId, memberId) => {
    const response = await api.delete(`/projects/${projectId}/members/${memberId}`)
    return unwrap(response)
  },

  // Get project activity
  getActivity: async (id) => {
    const response = await api.get(`/projects/${id}/activity`)
    return unwrap(response)
  },

  // Archive project
  archive: async (id) => {
    const response = await api.put(`/projects/${id}/archive`)
    return unwrap(response)
  },

  // Unarchive project
  unarchive: async (id) => {
    const response = await api.put(`/projects/${id}/unarchive`)
    return unwrap(response)
  },
}

File: crm-backend/src/main/java/com/arjun/crm/service/ProjectService.java
package com.arjun.crm.service;

import com.arjun.crm.dto.request.ProjectCreateRequest;
import com.arjun.crm.dto.request.ProjectUpdateRequest;
import com.arjun.crm.dto.response.ProjectResponse;
import com.arjun.crm.enums.ProjectStatus;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ProjectService {
    
    /**
     * Create a new project
     * @param request project creation details
     * @return created project response
     */
    ProjectResponse createProject(ProjectCreateRequest request);
    
    /**
     * Update project details
     * @param projectId project ID
     * @param request update details
     * @return updated project response
     */
    ProjectResponse updateProject(Long projectId, ProjectUpdateRequest request);
    
    /**
     * Delete project
     * @param projectId project ID to delete
     */
    void deleteProject(Long projectId);
    
    /**
     * Get project details
     * @param projectId project ID
     * @return project response
     */
    ProjectResponse getProject(Long projectId);
    
    /**
     * List projects by workspace
     * @param workspaceId workspace ID
     * @param status optional status filter
     * @param pageable pagination details
     * @return page of projects
     */
    Page<ProjectResponse> listProjectsByWorkspace(Long workspaceId, ProjectStatus status, Pageable pageable);
    
    /**
     * Archive project
     * @param projectId project ID to archive
     * @return archived project response
     */
    ProjectResponse archiveProject(Long projectId);
    
    /**
     * Unarchive project
     * @param projectId project ID to unarchive
     * @return unarchived project response
     */
    ProjectResponse unarchiveProject(Long projectId);
}
